framework/src/consumption/README.md

```diff
@@ -211,3 +211,38 @@ Using Client VPN Endpoint
 [example default](examples/opensearch-saml-clientvpn.lit.ts)
 
 
+[//]: # (consumption.opensearch-api)
+# OpenSearch API - Bring you own Opensearch cluster
+
+OpenSearch API client that allows to prepare the data or setup access roles for existing Opensearch clusters. The construct supports both OpenSearch provisioned clusters and OpenSearch Serverless collections.
+
+## Overview
+
+The construct leverages the [CDK Provider Framework](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources-readme.html#provider-framework) to deploy a custom resource to manage, and provide `addRoleMapping` and `callOpenSearchApi` methods. Both methods return the custom resource so that allows to enforce sequental execution of the API calls. By default all API calls will be executed simultaneously and are independent of each other. 
+
+[example OpenSearch API](examples/opensearch-api.lit.ts)
```

Comment:
Can you provide a dummy example without initializing the actual OS cluster with construct, but provide how to initialize the OS-api construct.